### ISSUE TITLE
Extremely slow performance when running tests with Xcode 15.0.1

### ISSUE THREAD

**Issue by mr-v** [actions/runner-images#8693]
### Description

For our test suite, the testing step (not including the build time) takes 20+ minutes. With Xcode 14.3, the testing part takes around 1 minute. The running time of individual tests looks okay.

Compared to Xcode 14.3:

1. There are now 50-60s long gaps in between the runs of each test suite.

```
[07:44:14]: ▸ All tests
[07:44:14]: ▸ Test Suite ...Tests.xctest started
[07:44:14]: ▸ ...Tests
[07:44:14]: ▸     ✔ tes... (1.023 seconds)
	↕ gap between the test suites
[07:45:13]: ▸ All tests
[07:45:13]: ▸ Test Suite ...Tests.xctest started
[07:45:13]: ▸ ...Tests
...
[07:45:13]: ▸     ✔ test... (0.010 seconds)
	↕ gap between the test suites
[07:46:02]: ▸ All tests
[07:46:02]: ▸ Test Suite ...Tests.xctest started
```
 With Xcode 14.3, the gap in logs is just 1-5s.

2. Near the end of the test run, there's a 10 minute time out.

> [08:02:52]: ▸ 2023-10-30 08:02:52.592 xcodebuild[39863:291538]  IDETestOperationsObserverDebug: Failure collecting diagnostics from simulator: Timed out after 600.0 seconds while waiting for a response from the invoked process

3. After that, the test run seems to be doing some work - it exits 4 minutes later.

> [08:02:57]: ▸ 2023-10-30 08:02:57.934 xcodebuild[39863:202553] [MT] IDETestOperationsObserverDebug: 1224.471 elapsed -- Testing started completed.
> [08:02:57]: ▸ 2023-10-30 08:02:57.934 xcodebuild[39863:202553] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
> [08:02:57]: ▸ 2023-10-30 08:02:57.934 xcodebuild[39863:202553] [MT] IDETestOperationsObserverDebug: 1224.471 sec, +1224.471 sec -- end
> [08:03:20]: ▸ Failing tests:
> ...
> [08:03:21]: ▸ ** TEST FAILED **
> [08:07:05]: ▸ Tests Passed: 0 failed, 274 total (12.554 seconds)
> [08:07:05]: Exit status: 65

(Regarding the test failures: see the note at the end of Environment section.)

### References

I'm unsure if you can do anything to improve this on your side, as it might be a problem on the Apple/Xcode side.

There's a similar report for CircleCI: [Severe performance problems with Xcode 15 - Build Environment / Mobile - CircleCI Discuss](https://discuss.circleci.com/t/severe-performance-problems-with-xcode-15/49205/40).

I reported this to Apple as FB13318262.

### Platforms affected

- [ ] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [X] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [X] macOS 13
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

20231024.2

### Is it regression?

If we're looking just at the Xcode 15, it's not a regression, as there were always performance issues with it.

### Expected behavior

Test suite execution time is comparable to Xcode 14.3.

### Actual behavior

In our experience, the tests which take around a minute with Xcode 14.3 on CI, take over 20 minutes with Xcode 15.0.1.

### Repro steps

Build and run the tests with Xcode 15.0.1.

It's an iOS app and we're using test plan for running the tests. I checked this with the unit test targets. There are tests for both app and frameworks.

**Environment**

Runner:  `macos-13-xlarge` (M1)
Simulator: `iPhone SE (3rd generation) iOS Simulator 17.0.1`
OS: macOS  13.6  22G120
Runner image: macos-13-arm64 20231024.2   Included Software: https://github.com/actions/runner-images/blob/macos-13-arm64/20231024.2/images/macos/macos-13-arm64-Readme.md

Note: for Xcode 14.3 we're using "iPhone 14 iOS 16.4" simulator, but I wouldn't expect this to make any difference outside of a few failures for tests which compare the look of UI.

**Comment by erik-bershel** [comment]
Hey @mr-v!
Generally it seems as a duplicate of this https://github.com/actions/runner-images/discussions/8651.
Shortly speaking it's a well known bug in XCode 15. Workaround which you may try placed [here](https://github.com/actions/runner-images/discussions/8651#discussioncomment-7393444).
In case you think this is a different problem, please call me so I can reopen the issue. Also feel free to open new issues or discussions in case of other problems.

**Comment by mr-v** [comment]
Hi @erik-bershel! Thanks for the pointer! (I found that discussion only after posting the issue here. 😅)
It does look related. However, I'd expect issues like this to be logged in the "Issues". Otherwise, it gives a wrong impression that there's no problem. Those performance problems are effectively blocking the adoption of Xcode 15 on CI for testing, so I think it's fair to acknowledge it and keep the issue open. Even if only for the purpose or redirecting folks to the https://github.com/actions/runner-images/discussions/8651 discussion and [the temporary workarounds](https://github.com/actions/runner-images/discussions/8651#discussioncomment-7393444).
